### PR TITLE
inkcut: init at 2.1.1

### PR DESCRIPTION
**Pull request by raboof**
###### Motivation for this change

Great application for driving vinyl cutter machines

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/enaml/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ply
+    kiwisolver
+    qtpy
+    sip
+    cppy
+    bytecode
+  ];
+
+  meta = with lib; {
+    homepage = https://github.com/frmdstryr/enaml;
```
Comment (erictapen) [inline pkgs/development/python-modules/enaml/default.nix:38]: I guess this is supposed to be `https://github.com/nucleic/enaml`?

**Review by erictapen — CHANGES_REQUESTED** [review]
This looks great. I checked that

- all packages have the correct licenses
- packaging style looks good
- meta.homepage matches src, with one exception
- inkcut builds and runs fine

File: pkgs/development/python-modules/enaml/default.nix (hunk 1 above)
Comment (erictapen) [reply pkgs/development/python-modules/enaml/default.nix:38]: Ah and could you quote all URLs? Unquoted URLs are deprecated and therefore discouraged: [RFC0045](https://github.com/NixOS/rfcs/blob/master/rfcs/0045-deprecate-url-syntax.md)

**Review by erictapen — APPROVED** [review]

**Review by jonringer — COMMENTED** [review]
To follow CONTRIBUTING.md, please do a commit per package addition.

technically, it says, separate commits into "logical units", but I consider a package addition to be a logical unit

**Comment by raboof** [comment]
(converting to draft since I want to look into cleaning up the qt-reactor packaging)

**Comment by raboof** [comment]
(split into commit-per-package and fixed up the qreactor package)

File: pkgs/applications/misc/inkcut/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    qtconsole
+    pyqt5
+  ];
+
+  # QtApplication.instance() does not work during tests?
+  doCheck = false;
+
+  postFixup = ''
+    wrapQtApp $out/bin/inkcut
+  '';
```
Comment (jonringer) [inline pkgs/applications/misc/inkcut/default.nix:35]: avoid double wrapping
```suggestion
  dontWrapQtApps = true;
  makeWrapperArgs = [ "''${qtWrapperArgs[@]}" ];
```

File: pkgs/development/python-modules/bytecode/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "bytecode";
+  version = "0.11.0";
+
+  src = fetchFromGitHub {
+    owner = "vstinner";
+    repo = pname;
+    rev = "${version}";
```
Comment (jonringer) [inline pkgs/development/python-modules/bytecode/default.nix:13]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/enaml/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchFromGitHub {
+    owner = "nucleic";
+    repo = pname;
+    rev = version;
+    sha256 = "1in5qa5j96qs3gsv8yaxs1l6dbm69xhzvc0pbzg0dd9kpqxfdy1j";
+  };
+
+  # qt bindings cannot be found during tests
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/enaml/default.nix:25]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

**Review by jonringer — COMMENTED** [review]
please apply testing comment to all related PRs

**Review by jonringer — COMMENTED** [review]
please address the failing builds:
```
[14 built (2 failed), 6 copied (464.3 MiB), 52.0 MiB DL]
error: build of '/nix/store/24jny6q4nkfdxjl5zwxv3vwf5lj6nw8i-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/92990
2 packages failed to build:
python27Packages.bytecode python27Packages.qreactor

9 packages built:
inkcut python37Packages.bytecode python37Packages.enaml python37Packages.enamlx python37Packages.qreactor python38Packages.bytecode python38Packages.enaml python38Packages.enamlx python38Packages.qreactor
```
appears to be from missing "aenum", you may need to do:
```
  propagateBuildInputs = [ ... ] ++ lib.optionals (pythonOlder "3.6") [ aenum];
```
example failure:
```
  ERROR: Could not find a version that satisfies the requirement aenum>=2.0; python_version < "3.6" (from bytecode==0.11.0) (from versions: none)
  ERROR: No matching distribution found for aenum>=2.0; python_version < "3.6" (from bytecode==0.11.0)
builder for '/nix/store/412b3qkcm3frbawrj99lvqx0hcd2ff7d-python2.7-qreactor-unstable-2018-09-29.drv' failed
```
since this also affect python3.5, I would like to have the optional dependencies included.

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

```https://github.com/NixOS/nixpkgs/pull/92990
9 packages built:
inkcut python37Packages.bytecode python37Packages.enaml python37Packages.enamlx python37Packages.qreactor python38Packages.bytecode python38Packages.enaml python38Packages.enamlx python38Packages.qreactor
```

**Comment by raboof** [comment]
Thanks for all the reviewing and pointers, I'm learning a lot ;)

**Pull request merged**